waitress/app/decorators.py

```diff
@@ -11,14 +11,15 @@ def guard(func):
     @wraps(func)
     def decorated_func(viewset, request, pk=None, *args, **kwargs):
         passphrase = request.POST.get('passphrase', None)
+        is_user_permitted = ((not request.user.is_anonymous) and (request.user.is_staff))
         exists = Passphrase.exists(passphrase)
-        if exists.status:
-            request.passphrase = exists.matched_list[0]
+        if exists.status or is_user_permitted:
+            request.passphrase = exists.matched_list[0] if exists.status else request.user.id
```

Comment:
Implicit branch statements. make it more pythonic by breaking it down.

Reply:
Removed the second condition. Thanks